velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -418,7 +418,7 @@ std::shared_ptr<const ParquetTypeWithId> ReaderBase::getParquetColumnInfo(
           std::move(children),
           curSchemaIdx,
           maxSchemaElementIdx,
-          columnIdx++,
+          columnIdx - 1, // was already incremented for leafTypePtr
```

Comment:
for my understanding, why here it's not ParquetTypeWithId::kNonLeaf but columnIdx - 1?

Reply:
the `else` statement is for the leaf type columns. If the column is a repeated type (like array), it would have the same column index as the column itself.